viper/src/main/java/com/techmagic/viper/base/BasePresenter.kt

```diff
@@ -73,6 +70,34 @@ abstract class BasePresenter<VIEW : View, ROUTER : Router> : Presenter {
         checkField("view", view)
     }
 
+    protected fun <T> call(
+            observable: Observable<T>,
+            onSuccess: (value: T) -> Unit,
+            onError: (throwable: Throwable) -> Unit = {
+                it.message?.let {
+                    view?.showErrorMessage(it)
+                }
+            }
+    ): Subscription {
+        val subscription = observable.subscribe({ onSuccess(it) }, { onError(it) })
+        compositeSubscription.add(subscription)
+        return subscription
```

Comment:
It was supposed to be a module which would be extracted to separate repository and probably published somewhere. It shouldn't have a dependency on RxJava especially on its first version.
Let's leave it as it is for the upcoming release, but please mark these methods to be extracted from this module later.

Reply:
Deprecated annotation would be just fine :)